ReorderableListView top padding causes clipping of content #79595

VrajGohil opened the pull request:
ReorderableListView, when using the padding parameter, does not pad the content of the list, but instead pads the entire list itself. This is not the expected result, and does not occur when padding the bottom either. It only occurs with top padding.

fix for this issue #79595 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

VrajGohil committed:
fix: ReorderableListView top padding #79595

VrajGohil committed:
fix: ReorderableListView top padding #79595

VrajGohil commented:
@googlebot I signed it!

PixelToast commented:
Only padding the bottom of the list appears to be intentional, looking at the comment above:
https://github.com/flutter/flutter/blob/8e653d11d846861e34ae8eaea0ecb413d3b0b1a1/packages/flutter/lib/src/material/reorderable_list.dart#L451-L454

cc @darrenaustin

VrajGohil committed:
Add test case: when additional padding applied

I was not able to find any test cases for addition padding applied. This commit adds the same for 'in vertical mode group'.

VrajGohil committed:
Fix a bug when header was added

pedromassangocode commented:
I can reproduce this issue.

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel master, 2.1.0-13.0.pre.444, on macOS 11.2.3 20D91 darwin-x64, locale en-AO)
    • Flutter version 2.1.0-13.0.pre.444 at /Users/pedromassango/Code/flutter_master
    • Framework revision 824b1ad3f8 (2 days ago), 2021-04-03 15:24:02 -0400
    • Engine revision e393800c63
    • Dart version 2.13.0 (build 2.13.0-196.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/pedromassango/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 54.1.4
    • Dart plugin version 203.7759

[✓] VS Code (version 1.54.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 11.2.3 20D91 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 89.0.4389.114

! Doctor found issues in 1 category.
```
</details>

darrenaustin reviewed (commented):
Thanks for the contribution. This looks pretty good, but doesn't address the case where the list is horizontal. Also, if we are going to put padding on the header itself, the `outerPadding` is really unnecessary.

As I want to get a fix out there as soon as possible, I just submitted a PR that will fix this for both vertical and horizontal versions of the list (#80251).  

I will go ahead and close this, but thanks again for providing a fix. We really appreciate it.

VrajGohil commented:
> Thanks for the contribution. This looks pretty good, but doesn't address the case where the list is horizontal. Also, if we are going to put padding on the header itself, the `outerPadding` is really unnecessary.
> 
> As I want to get a fix out there as soon as possible, I just submitted a PR that will fix this for both vertical and horizontal versions of the list (#80251).
> 
> I will go ahead and close this, but thanks again for providing a fix. We really appreciate it.

Thanks alot for reviewing. I'll make sure to cover all points in future contributions.
